internal/integration/stream_test_definitions.go

```diff
@@ -873,3 +873,29 @@ func StreamTestOutputOnlyOverride(getFn GetMessageFunc) StreamTestDefinition {
 		},
 	)
 }
+
+// StreamTestWithMetadata checks that a message with the given metadata will be received
+func StreamTestWithMetadata(metaFieldName, metaFieldValue string) StreamTestDefinition {
+	return namedStreamTest(
+		"can send to redis with stream name from metadata",
+		func(t *testing.T, env *streamTestEnvironment) {
+			t.Parallel()
+
+			tranChan := make(chan message.Transaction)
+			input, output := initConnectors(t, tranChan, env)
+			t.Cleanup(func() {
+				closeConnectors(t, input, output)
+			})
+
+			messageContent := `{"foo": "bar"}`
+			metaFieldVal := []string{metaFieldName, metaFieldValue}
+
+			require.NoError(t, sendMessage(env.ctx, t, tranChan, messageContent, metaFieldVal...))
+			received := receiveMessage(env.ctx, t, input.TransactionChan(), nil)
+
+			messageMatch(t, received, messageContent, metaFieldVal...)
+
+			// todo: add proper assert?
```

Comment:
Any good asserts here? My initial idea was to test the in/output configuration, but it is not available (for a good reason) and in theory it is enough to test that the message arrived. But then the test itself is kind of decoupled from what I am setting up on the other side in `integration_test.go` :/

Reply:
Now removed as no longer necessary to test this separately :)